src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -80,19 +78,21 @@ object Content {
    * @param hadoopConfiguration Hadoop configuration.
    * @param pathFilter Filter for accepting paths. The default filter is picked from spark
    *                   codebase, which filters out files like _SUCCESS.
-   * @param throwIfNotExists Throws FileNotFoundException if path is not found. Else creates a
-   *                         blank directory tree with no files.
    * @return Content object with Directory tree starting at root, and containing all leaf files
-   *         from "path" argument.
+   *         from "path" argument. If the given path does not exist, return Content object with
+   *         empty Directory object represents the path.
    */
   def fromDirectory(
```

Comment:
This is also called from `OptimizeAction`. Should we fail fast if the directory doesn't exist in that case? I think your original approach would handle it? (or we can add `failIfPathNotExists`)

Reply:
I think both ways are almost same in terms of call depth ?

Reply:
Maintainability wise, it seems better to handle here, but I will leave it to you.

Reply:
OptimizeAction also needs to ignore the error for its begin log - some tests failed without it and I added the fix after that - (https://github.com/microsoft/hyperspace/pull/453/commits/bcdfe9226ca19da1090690031aec490184169bfd).

I think it's okay as long as this is internal use only.

Reply:
But it seems weird that optimize() would create an empty directory, doesn't it?